arangod/CMakeLists.txt

```
@@ -1,3 +1,44 @@
+################################################################################
+## FAISS
+################################################################################
+
+find_package(Threads REQUIRED)
+find_package(ZLIB REQUIRED)
+
```

I would rather introduce the variable `USE_FAISS`  that is On by default, and if it a problem when building client tools jsut disable it and this won't run.
```suggestion
if (USE_FAISS)
find_package(Threads REQUIRED)
find_package(ZLIB REQUIRED)
...
endif()
```

---

Sorry, I do not understand. What should USE_FAISS do? Disable it globally?

---

I assume that will require a lot of changes with the source code.

---

And I definitely do not want to have different build directory for arangod and arangosh.

---

If I understand the issue is that building one of the client tools i.e. arangosh requires all libraries in 3rdParty to be present even when not used.
 If you want to just avoid that when you know the code compiling does not need it just add the global variable  `USE_FAISS` and disable it in the configuration `cmake -DUSE_FAISS=Off` when building arangosh, but make it `On` by default. That does not require any changes except adding the if clause and setting the default value of `USE_FAISS` to `On`

---

@jbajic CMake does not work that way. If you include FAISS on Top-Level it will add too much to "arangosh". Therefore simply move it to "arangod/CMakeLists.txt". This will solve the issue.

---

I see we are doing the same for `LIBUNWIND`, and I managed to compile arangosh by disabling faiss. But if this causes issues with Conda, then go ahead.

---

Yes, it mostly arises if you compile with Conda. But you have a point there with RocksDB as well. If this is only used by arangod we should move it there as well.